pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -328,7 +344,7 @@ func (o *MustGatherOptions) Run() error {
 				return err
 			}
 			for _, node := range nodes.Items {
-				pod, err := o.Client.CoreV1().Pods(ns.Name).Create(context.TODO(), o.newPod(node.Name, image), metav1.CreateOptions{})
+				pod, err := o.Client.CoreV1().Pods(ns.Name).Create(context.TODO(), o.newPod(node.Name, image, hasMaster), metav1.CreateOptions{})
```

Comment:
Before we start adding weird edge cases, have you considered using `--node-selector=node-role.kubernetes.io/worker` instead?

Reply:
I wouldn't look at Openshift hosted clusters powered by hypershift as a weird use case but rather as the only/default use case for Red Hat Openshift managed services offering and the increasing choice for self-hosted multi-cluster offering.

The intend was to avoid end user to have to set an additional flag for it work ootb in any of the scenarios above.
Whether this is the right impl approach to satisfy the use case I have not opinions on this code base, so feel free to close and request a deeper discussion. I'm curious why we run in a master by default?

Reply:
> I'm curious why we run in a master by default?

We care about what's going on masters only, even though parts of the data we scrape is reachable through API, significant portion is scraping data directly from the OS running the control plane components. That's why it was natural for us to force running them on masters.